lib/improver/spotdata/framework.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""The framework for site specific post-processing."""
+
+import os
+import multiprocessing as mp
+from iris.cube import CubeList
+
+from improver.spotdata.read_input import (Load,
+                                          get_method_prerequisites,
+                                          read_config)
+from improver.spotdata.neighbour_finding import PointSelection
+from improver.spotdata.extract_data import ExtractData
+from improver.spotdata.write_output import WriteOutput
+from improver.spotdata.ancillaries import get_ancillary_data
+from improver.spotdata.extrema import ExtractExtrema
+from improver.spotdata.site_data import ImportSiteData
+from improver.spotdata.times import get_forecast_times
+from improver.spotdata.common_functions import (construct_neighbour_hash,
+                                                datetime_constraint,
+                                                extract_cube_at_time,
+                                                extract_ad_at_time)
+
+
```

Comment:
No. It was all written rather standalone. It probably needs to be restructured in a more plugin style way that can be controlled from the suite. I'm still waiting on a better understanding of how exactly it's going to be used before I start rearranging it. I have a meeting on Friday that may clarify things, but for now it remains a bit of an improver oddity.